sdk-lib/src/main/java/cash/z/ecc/android/sdk/block/CompactBlockProcessor.kt

```diff
@@ -306,35 +344,34 @@ class CompactBlockProcessor(
      * prevHash value matches the preceding block in the chain.
      *
      * @param lastScanRange the range to be validated and scanned.
-     *
-     * @return error code or [ERROR_CODE_NONE] when there is no error.
-     */
-    private suspend fun validateAndScanNewBlocks(lastScanRange: IntRange): Int = withContext(IO) {
-        setState(Validating)
-        var error = validateNewBlocks(lastScanRange)
-        if (error == ERROR_CODE_NONE) {
-            // in theory, a scan should not fail after validation succeeds but maybe consider
-            // changing the rust layer to return the failed block height whenever scan does fail
-            // rather than a boolean
-            setState(Scanning)
-            val success = scanNewBlocks(lastScanRange)
-            if (!success) throw CompactBlockProcessorException.FailedScan()
-            else {
-                setState(Scanned(lastScanRange))
+     */
+    private suspend fun validateAndScanNewBlocks(lastScanRange: ClosedRange<BlockHeight>?): BlockProcessingResult =
+        withContext(IO) {
+            setState(Validating)
+            val error = validateNewBlocks(lastScanRange)
+            if (error == BlockProcessingResult.Success) {
+                // in theory, a scan should not fail after validation succeeds but maybe consider
+                // changing the rust layer to return the failed block height whenever scan does fail
+                // rather than a boolean
+                setState(Scanning)
+                val success = scanNewBlocks(lastScanRange)
+                if (!success) {
+                    throw CompactBlockProcessorException.FailedScan()
+                } else {
+                    setState(Scanned(lastScanRange))
+                }
             }
-            ERROR_CODE_NONE
-        } else {
+
```

Comment:
I think there's a dangling vertical space here

Reply:
I can't find this anymore